protocols/v2/noise-sv2/README.md

```diff
@@ -0,0 +1,36 @@
+
+<div align="center">
+<h1>Noise-SV2</h1>
+<p>
+    A noise protocol implemenation for Stratum V2
+</p>
+<p>
+    <a href="https://github.com/stratum-mining/stratum/blob/main/LICENSE.md"><img alt="CC0 1.0 Universal Licensed" src="https://img.shields.io/badge/license-CC0--1.0-blue.svg"/></a>
+    <a href="https://app.codecov.io/gh/stratum-mining/stratum/tree/main/protocols%2Fv2%2Fnoise-sv2">
+    <img src="https://codecov.io/gh/stratum-mining/stratum/branch/main/graph/badge.svg" alt="codecov">
+</a>
+    <a href="https://blog.rust-lang.org/2023/12/28/Rust-1.75.0.html"><img alt="Rustc Version 1.75.0+" src="https://img.shields.io/badge/rustc-1.75.0%2B-lightgrey.svg"/></a>
+    <a href="https://crates.io/crates/noise_sv2"><img alt="crate.io" src="https://img.shields.io/crates/v/noise_sv2.svg"/></a>
+    <a href="https://docs.rs/noise_sv2"><img alt="docs" src="https://docs.rs/noise_sv2/badge.svg"/></a>
+  </p>
+</div>
```

Comment:
I think HTML for this header is overkill

we already introduced a header on `const_sv2`,  we should aim to follow the same formatting as much as possible

vanilla markdown can do the job here, just like we did in `const_sv2`

Reply:
I like the extra badges we added here (a few more than `const_sv2`), but:

- the license header is rendering as `CC0-1.0`, which is incorrect... we should either fix this or get rid of this badge
- MSRV is baked into the `README.md`, which could easily get outdated in case we bump MSRV globally and forget to update the badges... so IMO we should either:
  1. find a way to get this to render this badge automatically with the current MSRV 
  2. just not add this badge

tbh 2. feels more appealing, not sure how realistic it is to achieve 1. without getting hacky/dirty

Reply:
Ooohh. Idk...the badges here look pretty professional. I actually prefer this format over what we currently have in that `const-sv2` PR `README.md`. Shall we get @GitGab19 to break the tie?

Reply:
I'm not against having HTML, as long as all other crates also follow the same pattern.

I am against having each crate doing its own thing though

Also, my other two remakrs about License and and MSRV badges remain, regardless of HTML.

Reply:
Oh, yeah. I agree we should not use HTML and stick with markdown. Unless there is a specific feature that HTML gives us that we want that markdown does not support, which I do not think is the case here.

For the license and the code coverage badges, perhaps we just include those on the `README.md` in the repo root?

Reply:
I am removing HTML rendering in the next push. However, since these crates will be standalone, it is essential to include information about their licensing and code coverage, as the community may want to be informed about these aspects.

Reply:
Updated the licensing to reference MIT/Apache-2.0. @Plebhash, most of the public crates I've seen mention the supported MSRV. Even though we may bump the MSRV in the future, users should be informed about the version they are using, either through the documentation or the README. All major Rust projects in our ecosystem include this information.

Reply:
> Ooohh. Idk...the badges here look pretty professional. I actually prefer this format over what we currently have in that `const-sv2` PR `README.md`. Shall we get @GitGab19 to break the tie?

I like these badges, and I also think they should all be there. I saw @Shourya742 removed the HTML, so I would say we just need to specify them in the github discussion related to documentation stardards. After that, I will adapt the `README` I've created for `const_sv2` to match them

Reply:
Added here: 693f638f